src/util.ts

```diff
@@ -58,11 +63,6 @@ export async function synthApp(command: string, outdir: string, stdout: boolean,
     },
   });
 
-  if (!await fs.pathExists(outdir)) {
-    console.error(`ERROR: synthesis failed, app expected to create "${outdir}"`);
-    process.exit(1);
```

Comment:
I understand where your coming from with having `synth` not fail in this case (because manifests were actually created). However, from the CLIs perspective, synthesis is not only creating manifests, there are additional CLI features (like validations) that rely on the fact this directory exists, and so from its POV, synthesis is not successful. This does unfortunately mean that, if you define `outdir` in your app, you must also change the `output` property in the `cdk8s.yaml` configuration file. 

Note that if you do it the other way around, i.e set `output: out` in `cdk8s.yaml`, and leave the `App` instance with its defaults, everything will work as expected.